src/Uno.UWP/Storage/StorageFile.Base.cs

```diff
@@ -0,0 +1,102 @@
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+using Buffer = Windows.Storage.Streams.Buffer;
+
+namespace Windows.Storage
+{
+	public sealed partial class StorageFile
+	{
+		private abstract class ImplementationBase
+		{
+			protected ImplementationBase(string path)
+				=> Path = path;
+
+			public void InitOwner(StorageFile owner)
+				=> Owner = owner; // Lazy initialized to avoid delegate in StorageFile ctor
+
+			protected StorageFile Owner { get; private set; } = null!; // Should probably be private
```

Comment:
Yeah it's weird but this property is actually [initialized in the `StorageFile` constructor](https://github.com/unoplatform/uno/pull/4232/files/5c878903f0ae7825dca1911e4625f347ea088bd6#diff-c37c227b6b5f1bebb4726a16c77e398bR44) before any usage, and since this is a private nested class it really impossible to that it's `null`. 

I agree that it's based on the current usage and is a bit risky in case refactor, I thought it was better (for code readability and performance) to avoid the cost of assuming the possibility to be null, or have to deal with a delegate to lazy construct the implementation with the `StorageFile` instance as parameter.